derivers.lua

```diff
@@ -511,7 +809,145 @@ local diff = {
 			variants_checks[left.kind](left, right)
 		end
 
-		idx["diff"] = diff_fn
+		idx.diff = diff_fn
+		t.derived_diff = true
+	end,
+	foreign = function()
+		error("can't derive diff for a foreign type")
```

Comment:
change to "can't derive diff()" for more clarity

Reply:
resolved in #95